Watchman.Discord/Areas/Protection/Strategies/SpamPunishmentStrategy.cs

```diff
@@ -3,24 +3,26 @@
 using Devscord.DiscordFramework.Framework.Commands.AntiSpam.Models;
 using Serilog;
 using Watchman.Discord.Areas.Protection.Services;
+using Watchman.Discord.Areas.Users.Services;
 
 namespace Watchman.Discord.Areas.Protection.Strategies
 {
     public class SpamPunishmentStrategy : ISpamPunishmentStrategy
     {
         private readonly IPunishmentsCachingService _punishmentsCachingService;
+        private readonly IWarnsService _warnsService;
 
-        public SpamPunishmentStrategy(IPunishmentsCachingService punishmentsCachingService)
+        public SpamPunishmentStrategy(IPunishmentsCachingService punishmentsCachingService, IWarnsService warnsService)
         {
             this._punishmentsCachingService = punishmentsCachingService;
+            this._warnsService = warnsService;
         }
 
-        public Punishment GetPunishment(ulong userId, SpamProbability spamProbability)
+        public Punishment GetPunishment(ulong userId, ulong serverId, SpamProbability spamProbability)
         {
             var takeFromTime = DateTime.Now.AddHours(-12);
-            var warnsCount = this._punishmentsCachingService.GetUserWarnsCount(userId, takeFromTime);
+            var warnsCount = _warnsService.GetWarnsCount(userId, serverId, takeFromTime);              
```

Comment:
druga rzecz - w antispam szybkość działania jest bardzo ważna, dlatego pobieranie warns z bazy przy każdej wiadomości będzie generować duże opóźnienia w obsłudze wiadomości
pamiętaj że jak ktoś spamuje to czas na obsługę wiadomości wynosi kilkadziesiąt ms, a najlepiej byłoby zejść do kilkunastu
pomyśl nad pobieraniem tych warns przez punishmentsCachingService (oczywiście powinien on brać pod uwagę te warny nadane przez admina)
możesz na przykład okresowo pobierać z bazy warny, albo może nawet lepiej - powiadamiać o nowym warnie ten serwis

Reply:
zrobiłem że WarnsService zapisuje do bazy oraz do cache